File: restring/src/test/java/com/ice/restring/RestringLayoutInflaterTest.java
package com.ice.restring;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.ArgumentCaptor;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.mockito.stubbing.Answer;
import org.robolectric.RobolectricTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.atLeastOnce;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@RunWith(RobolectricTestRunner.class)
@Config(sdk = {16, 19, 21, 23, 24, 26})
public class RestringLayoutInflaterTest {

    @Mock private ViewTransformerManager transformerManager;
    private RestringLayoutInflater restringLayoutInflater;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
        when(transformerManager.transform(any(), any())).thenAnswer((Answer<View>) invocation ->
                invocation.getArgument(0)
        );
        RuntimeEnvironment.application.setTheme(R.style.Theme_AppCompat);
        restringLayoutInflater = new RestringLayoutInflater(
                LayoutInflater.from(RuntimeEnvironment.application),
                RuntimeEnvironment.application,
                transformerManager,
                false
        );
    }

    @Test
    public void shouldTransformViewsOnInflatingLayouts() {
        ViewGroup viewGroup = (ViewGroup) restringLayoutInflater.inflate(R.layout.test_layout, null, false);

        ArgumentCaptor<View> captor = ArgumentCaptor.forClass(View.class);
        verify(transformerManager, atLeastOnce()).transform(captor.capture(), any());
        for (int i = 0; i < viewGroup.getChildCount(); i++) {
            View child = viewGroup.getChildAt(i);
            captor.getAllValues().contains(child);
        }
    }
}

File: restring/src/test/java/com/b3nedikt/restring/RestringLayoutInflaterTest.kt
package com.b3nedikt.restring

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.test.core.app.ApplicationProvider
import com.nhaarman.mockitokotlin2.*
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [16, 19, 21, 23, 24, 26])
class RestringLayoutInflaterTest {

    private lateinit var transformerManager: ViewTransformerManager
    private lateinit var restringLayoutInflater: RestringLayoutInflater

    @Before
    fun setUp() {

        transformerManager = mock()

        whenever(transformerManager.transform(any(), any()))
                .thenAnswer{LinearLayout(ApplicationProvider.getApplicationContext())}

        ApplicationProvider.getApplicationContext<Context>().setTheme(R.style.Theme_AppCompat)
        restringLayoutInflater = RestringLayoutInflater(
                LayoutInflater.from(ApplicationProvider.getApplicationContext()),
                ApplicationProvider.getApplicationContext(),
                transformerManager,
                false
        )
    }

    @Test
    fun shouldTransformViewsOnInflatingLayouts() {
        val viewGroup = restringLayoutInflater.inflate(R.layout.test_layout, null, false) as ViewGroup

        val captor = argumentCaptor<View>()
        verify(transformerManager, atLeastOnce()).transform(captor.capture(), any())
        for (i in 0 until viewGroup.childCount) {
            val child = viewGroup.getChildAt(i)
            captor.allValues.contains(child)
        }
    }
}